File: py/throttle.py
import threading

class Limiter:
  pass

def throttle(f, interval):
  limiter = Limiter()
  limiter.limited = False
  def wrapped(*args):
    if limiter.limited:
      return
    else:
      limiter.limited = True
      def reset():
        limiter.limited = False
      threading.Timer(interval/1000, reset).start()
      return f(*args)
  return wrapped

File: rb/throttle.rb
module Throttle

  def self.throttle interval, f=nil, &block
    raise("Either a lambda or a block must be given") unless f || block_given?
    f ||= block
    limited = false
    -> (*args) do
      return nil if limited
      limited = true
      Thread.new do
        sleep(interval / 1000)
        limited = false
      end
      f.(*args)
    end
  end

end
